woocommerce/woocommerce-gutenberg-products-block
# Update `AddressForm` and related components to TypeScript

Converted `AddressForm`, `prepare-address-fields`, `BillingCountryInput`, `ShippingCountryInput`, `CountryInput`, `BillingStateInput`, `ShippingStateInput`, and `StateInput` to TypeScript.

This includes a cherry-picked commit from #4361 - so feel free to ignore the changes in `assets/js/settings/shared/utils.ts` for the purposes of this PR

### How to test the changes in this Pull Request:

1. Smoke test cart and checkout functionality.
2. Ensure that disallowed countries are not selectable in the checkout form.
3. Ensure address can be submitted and it is sent to the API correctly and stored correctly in the order.
